src/containers/Storage/Storage.tsx

```diff
@@ -134,12 +157,8 @@ export const Storage = ({database, viewContext, nodeId, groupId, pDiskId}: Stora
         [searchValue, groups],
     );
 
-    const [nodesSort, handleNodesSort] = useTableSort(nodesSortParams, (params) =>
-        setNodeSort(params as NodesSortParams),
-    );
-    const [groupsSort, handleGroupsSort] = useTableSort(groupsSortParams, (params) =>
-        setGroupSort(params as StorageSortParams),
-    );
+    const [nodesSort, handleNodesSort] = useStorageSort(nodesSortParams, setNodeSort);
```

Comment:
Maybe istead of nesting hooks we could do more straightforward:

```
       function fromTableSort<T extends NodesSortParams | StorageSortParams>(
        sort: SortOrder[] | undefined,
    ): T | undefined {
        if (!sort?.[0]) {
            return undefined;
        }
    
        return {
            sortValue: sort[0].columnId,
            sortOrder: sort[0].order,
        } as T;
    }

    const [nodesSort, handleNodesSort] = useTableSort({
        initialSortColumn: nodesSortParams.sortValue,
        initialSortOrder: nodesSortParams.sortOrder,
        multiple: false,
        onSort: (sort) => setNodeSort(fromTableSort<NodesSortParams>(sort)),
    });

    const [groupsSort, handleGroupsSort] = useTableSort({
        initialSortColumn: groupsSortParams.sortValue,
        initialSortOrder: groupsSortParams.sortOrder,
        multiple: false,
        onSort: (sort) => setGroupSort(fromTableSort<StorageSortParams>(sort)),
    });
 ```

Reply:
Current version works as before, I don't see the reason to make it beautiful, since you are deleting it anyway in #1693